### PR TITLE
Implement AssetDepositLocal, AssetWithdrawLocal

### PR DESCRIPTION
**Pull request by patrickxb**

### CODE REVIEW

File: go/stellar/stellarsvc/anchor.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+func (a *anchorInteractor) Withdraw(mctx libkb.MetaContext) (stellar1.AssetActionResultLocal, error) {
+	if err := a.checkAsset(mctx); err != nil {
+		return stellar1.AssetActionResultLocal{}, err
+	}
+	u, err := a.checkURL(mctx, "withdraw")
+	if err != nil {
+		return stellar1.AssetActionResultLocal{}, err
+	}
+
+	// TODO: `type` is supposedly optional, but actually required...so figure it out.
```
Comment (xgess) [inline go/stellar/stellarsvc/anchor.go]: on https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0006.md it looks like it's required for withdrawals and optional for deposits. i dont understand though. it doesn't look like we're including it. and this says account is required for withdraws, but it doesn't look like we're sending that either.

File: go/stellar/stellarsvc/anchor_test.go [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+			continue
+		}
+		if test.DepositExternalURL != "" && res.ExternalUrl == nil {
+			t.Errorf("valid test %d [%s] deposit: result external url field is nil, expected %s", i, test.Name, test.DepositExternalURL)
+			continue
+		}
+		if res.ExternalUrl != nil {
+			if test.DepositExternalURL != *res.ExternalUrl {
+				t.Errorf("valid test %d [%s] deposit: result external url field %s, expected %s", i, test.Name, *res.ExternalUrl, test.DepositExternalURL)
+
```
Comment (xgess) [inline go/stellar/stellarsvc/anchor_test.go]: extra new line

File: go/stellar/stellarsvc/anchor_test.go [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+			continue
+		}
+		if test.WithdrawExternalURL != "" && res.ExternalUrl == nil {
+			t.Errorf("valid test %d [%s] withdraw: result external url field is nil, expected %s", i, test.Name, test.WithdrawExternalURL)
+			continue
+		}
+		if res.ExternalUrl != nil {
+			if test.WithdrawExternalURL != *res.ExternalUrl {
+				t.Errorf("valid test %d [%s] withdraw: result external url field %s, expected %s", i, test.Name, *res.ExternalUrl, test.WithdrawExternalURL)
+
```
Comment (xgess) [inline go/stellar/stellarsvc/anchor_test.go]: here too

File: go/stellar/stellarsvc/anchor.go (hunk 1 above)
Comment (patrickxb) [reply go/stellar/stellarsvc/anchor.go]: I asked and was told it was optional several days ago.  There is a PR to change the spec:

https://github.com/stellar/stellar-protocol/pull/358

but all the anchors i've tried currently require it.

i'm going to fix this now by getting additional info from stellard, but i wanted to get the deposit flow in for UI testing.

Comment (xgess) [reply go/stellar/stellarsvc/anchor.go]: ahh ahh this is what you were saying earlier. got it.

File: go/stellar/stellarsvc/anchor_test.go [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+// mockAnchorUSDTransferGet is an httpGetClient func that returns a stored result
+// for TRANSFER_SERVER/deposit and TRANSFER_SERVER/withdraw
+func mockAnchorUSDTransferGet(mctx libkb.MetaContext, url string) (int, []byte, error) {
+	parts := strings.Split(url, "?")
+	switch parts[0] {
+	case "https://api.anchorusd.com/transfer/deposit":
+		return http.StatusForbidden, []byte(depositBody), nil
+	case "https://api.anchorusd.com/transfer/withdraw":
+		return http.StatusForbidden, []byte(withdrawBody), nil
```
Comment (xgess) [inline go/stellar/stellarsvc/anchor_test.go]: it looks like there's nothing that tests the http.StatusOK branch of logic. do we not expect these to happen / do they not actually exist in practice?

Comment (patrickxb) [reply go/stellar/stellarsvc/anchor_test.go]: I'll add a test, that was an oversight.  But I don't expect them to happen.

Comment (xgess) [reply go/stellar/stellarsvc/anchor_test.go]: yup. that makes sense. cool.

**Pull request merged**